sw/airborne/modules/flight_plan_guided/flight_plan_guided.c

```diff
@@ -0,0 +1,352 @@
+bool SpeedRotateToHeading(float rotation_rate, float heading)
+{
+  guidance_h_set_guided_heading_rate(rotation_rate);
+
+  if (fabs(wrapToPi_guided(heading) - stateGetNedToBodyEulers_f()->psi) < 0.1) {
+    guidance_h_set_guided_heading(heading);
+    return false;
+  }
+  return true;
+}
+
+/********************************************//**
+ *  Blocking conditions
+ ***********************************************/
+
+/**
+ * WaitUntilAltitude: Blocks while waiting until a certain altitude is reached
+ * @param[in] altitude in [m] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilAltitude(float altitude)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+  if (stateGetPositionEnu_f()->z < altitude) { return true; }
+
+  return false;
+}
+
+
+/**
+ * WaitUntilAltitude: Blocks while waiting until a certain altitude is reached
+ * @param[in] altitude in [m] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilGround(void)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+  if (stateGetPositionEnu_f()->z > 0.1) { return true; }
+
+  return false;
+}
+
+/**
+ * WaitUntilSpeedOrAltitude: Blocks while waiting until a certain altitude or speed is reached
+ * @param[in] fail_altitude, wait for altitude in [m] (positive z-axis is up)
+ * @param[in] speed, speed in [m/s] (positive z-axis is up)
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitUntilSpeedOrAltitude(float speed, float altitude)
+{
+  if (autopilot.mode != AP_MODE_GUIDED) { return true; }
+
+  if (stateGetPositionEnu_f()->z < altitude && stateGetSpeedEnu_f()->z < speed) { return true; }
+
+  return false;
+}
+
+/**
+ * WaitforHeading: Blocks untill wanted heading is achieved
+ * @param[in] heading, waits untill MAV reached certain heading [rad] (clockwise is positive)
+ *  @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ *
+ * TODO: Wrap the angle around -pi and pi
+ */
+bool WaitforHeading(float heading)
+{
+  if (fabs(wrapToPi_guided(heading) - stateGetNedToBodyEulers_f()->psi) < 0.1) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * WaitForHMode: Blocks until specified horizontal mode is achieved (handy for if you want to start the
+ * flightplan with a flip of the switch)
+ * @param[in] mode, number of the specific horizontal mode that you want to wait for
+ * @param[return] Boolean for flight plan to keep running (true) or to just run once (false)
+ */
+bool WaitForHMode(uint8_t mode)
+{
+  if (guidance_h.mode == mode) {
+    return false;
+  }
+  return true;
+}
+
+
+/********************************************//**
+ *  Timer functions
+ ***********************************************/
+
+/**
+ * ResetSpecialTimer: Reset a special timer, that works outside of the block and stage time
```

Comment:
If you want it like this just change the comment to not say it "works outside of the block and stage time".